test/integration/test_create_dss_docker_image.py

```diff
@@ -167,3 +190,104 @@ def test_docker_socket_on_host_touched(dss_container_context, fake_docker_socket
         wait_for_socket_access(container)
 
     assert stat_before == socket.stat()
+
+
+def assert_exec_run(container: Container, command: str, **kwargs) -> str:
+    """
+    Execute command in container and verify success.
+    """
+    exit_code, output = container.exec_run(command, **kwargs)
+    output = output.decode("utf-8").strip()
+    assert exit_code == 0, output
+    return output
+
+
+class SocketInspector:
+    def __init__(self, context_provider, socket_on_host: Path):
+        self._context_provider = context_provider
+        self.socket_on_host = socket_on_host
+        self._container = None
+
+    @contextmanager
+    def context(self):
+        with self._context_provider(self.socket_on_host) as container:
+            wait_for_socket_access(container)
+            self._container = container
+            yield self
+            self._container = None
+
+    def run(self, command: str, **kwargs) -> str:
+        return assert_exec_run(self._container, command, **kwargs)
+
+    def numeric_gid(self, group_entry: str) -> int:
+        """group_entry is "ubuntu:x:971:", for example"""
+        return int(group_entry.split(':')[2])
+
+    def get_gid(self, group_name: str) -> int:
+        output = self.run(f"getent group {group_name}")
+        return self.numeric_gid(output)
+
+    def get_unassigned_gid(self) -> int:
+        """
+        Return a new gid, that is not used for any other group, yet.
+        """
+        gid = 0
+        for line in self.run("cat /etc/group").splitlines():
+            if not line.startswith(f"nogroup:"):
+                gid = max(gid, self.numeric_gid(line))
+        return gid + 1
+
+    def assert_jupyter_member_of(self, group_name: str):
+        output = self.run(f"getent group {group_name}")
+        members = output.split(":")[3].split(",")
+        assert "jupyter" in members
+
+    def assert_write_to_socket(self):
+        signal = f"Is there anybody out there {datetime.now()}?"
+        self.run(
+            f'bash -c "echo {signal} > {DOCKER_SOCKET_CONTAINER}"',
+            user="jupyter")
+        assert signal == self.socket_on_host.read_text().strip()
+
+
+@pytest.fixture
+def socket_inspector(dss_container_context, accessible_file):
+    yield SocketInspector(dss_container_context, accessible_file)
+
+
+class GroupChanger:
+    def __init__(self, context_provider):
+        self._context_provider = context_provider
+
+    def chgrp(self, gid: int, path_on_host: Path):
+        path_in_container = "/mounted"
+        with self._context_provider(path_on_host, path_in_container) as container:
+            assert_exec_run(container, f"chgrp {gid} {path_in_container}")
+
+
+@pytest.fixture
+def group_changer(ubuntu_container_context):
+    return GroupChanger(ubuntu_container_context)
+
+
+def test_write_socket_known_gid(socket_inspector, group_changer):
+    with socket_inspector.context() as inspector:
+        gid = inspector.get_gid("ubuntu")
```

Comment:
I am not sure who good it is to hard-code the ubuntu group

Reply:
I wanted to select a group that was not present on my host machine.
Can you imagine a way to pick a group without hard-coding?

Reply:
Enhance ansible to add environment variable into the container with name of group (currently `ubuntu`).